calendar/index.js

```diff
@@ -0,0 +1,273 @@
+            this.clearInputFields();
+            clearSessionData(this.startStorageKey, this.endStorageKey);
+        } else if (this.state.checkin && this.state.checkout) {
+            this.setRangeInputFields();
+        } else if (this.state.checkin) {
+            this.setSingleInputField();
+        }
+    }
+
+    setRangeInputFields() {
+        const checkinDate = new Date(this.state.checkin);
+        const checkoutDate = new Date(this.state.checkout);
+    
+        this.inputElement.value = `${formatDate(checkinDate, this.dateLocale)} - ${formatDate(checkoutDate, this.dateLocale)}`;
+        this.startInput.value = checkinDate.toISOString();  // Save as ISO string
+        this.endInput.value = checkoutDate.toISOString();   // Save as ISO string
+    
+        saveToSessionStorage(this.startStorageKey, this.startInput.value);
+        saveToSessionStorage(this.endStorageKey, this.endInput.value);
+    }
+    
+
+    setSingleInputField() {
+        const checkinDate = new Date(this.state.checkin);
+    
+        this.inputElement.value = formatDate(checkinDate, this.dateLocale);
+        this.startInput.value = checkinDate.toISOString();  // Save as ISO string
+        this.endInput.value = '';
+    
+        saveToSessionStorage(this.startStorageKey, this.startInput.value);
+        clearSessionData(this.endStorageKey);
+    }
+    
+    clearInputFields() {
+        this.inputElement.value = '';
+        this.startInput.value = '';
+        this.endInput.value = '';
+    }
+
+    resetDates() {
+        this.state.checkin = null;
+        this.state.checkout = null;
+        clearHighlights(this.dayButtons, [CLASS_NAMES.checkin, CLASS_NAMES.checkout, CLASS_NAMES.inRange]);
+        this.updateInputField(true);
+    }
+
+    highlightDayForTime(time, className) {
+        const dayButton = this.dayButtons.find(button => parseInt(button.dataset.time) === time);
+        if (dayButton) highlightButton(dayButton, className);
+    }
+
+    highlightRange(startTime, endTime) {
+        this.dayButtons.forEach(button => {
+            const time = parseInt(button.dataset.time);
+            if (time > startTime && time < endTime) {
+                highlightButton(button, CLASS_NAMES.inRange);
+            }
+        });
+    }
+
+    toggleCalendar() {
+        this.calendarVisible = !this.calendarVisible;
+        toggleVisibility(this.panel, this.calendarVisible);
+
+        if (this.calendarVisible) {
+            document.addEventListener('keydown', this.handleEscKey.bind(this));
+        } else {
+            document.removeEventListener('keydown', this.handleEscKey.bind(this));
+        }
+    }
+
+    handleEscKey(event) {
+        if (event.key === 'Escape') toggleVisibility(this.panel, false);
+    }
+
+    loadFromSessionStorage() {
+        const storedCheckin = getFromSessionStorage(this.startStorageKey);
+        const storedCheckout = getFromSessionStorage(this.endStorageKey);
+    
+        if (storedCheckin) {
```

Comment:
@samuelsilvapt This should be a function since your are replicating the same thing below